android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/identitycredential/SelectAccountDialog.kt

```diff
@@ -0,0 +1,159 @@
+) {
+    Column(
+        modifier = modifier.fillMaxWidth(),
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.padding(16.dp),
+        ) {
+            provider.icon?.base64PngToBitmap()?.asImageBitmap()?.let {
+                Image(
+                    bitmap = it,
+                    contentDescription = null,
+                    contentScale = ContentScale.FillWidth,
+                    modifier = Modifier
+                        .padding(end = 4.dp)
+                        .size(16.dp),
+                )
+            }
+            Text(
+                text = stringResource(
+                    id = R.string.mozac_feature_prompts_identity_credentials_choose_account_for_provider,
+                    provider.name,
+                ),
+                style = TextStyle(
+                    fontSize = 16.sp,
+                    lineHeight = 24.sp,
+                    color = PhotonColors.DarkGrey90,
+                    letterSpacing = 0.15.sp,
+                ),
+            )
+        }
+
+        accounts.forEach { account ->
+            AccountItem(account = account, onClick = onAccountClick)
+        }
+
+        Spacer(modifier = Modifier.height(16.dp))
+    }
+}
+
+@Composable
+private fun AccountItem(
+    account: Account,
+    modifier: Modifier = Modifier,
+    onClick: (Account) -> Unit,
+) {
+    ListItem(
+        title = account.name,
+        description = account.email,
+        modifier = modifier,
+        onClick = { onClick(account) },
+    ) {
+        account.icon?.base64PngToBitmap()?.asImageBitmap()?.let { bitmap ->
+            Image(
+                bitmap = bitmap,
+                contentDescription = null,
+                contentScale = ContentScale.FillWidth,
+                modifier = Modifier
+                    .padding(horizontal = 16.dp)
+                    .size(32.dp),
+            )
+        } ?: Spacer(
+            Modifier
+                .padding(horizontal = 16.dp)
+                .width(32.dp),
+        )
+    }
+}
+
+@Composable
+@Preview(name = "Provider with no favicon")
+private fun AccountItemPreview() {
+    AccountItem(
+        modifier = Modifier.background(Color.White),
+        account = Account(
+            0,
+            "user@mozilla.com",
+            "User",
+            USER_PICTURE,
+        ),
+        onClick = {},
+    )
+}
+
+@Composable
+@Preview
+private fun SelectAccountDialogPreview() {
+    SelectAccountDialog(
+        provider = Provider(0, GOOGLE_FAVICON, "Google", "google.com"),
+        accounts = listOf(
+            Account(
+                0,
+                "user@mozilla.com",
+                "User",
+                USER_PICTURE,
+            ),
+            Account(
+                1,
+                "user2@mozilla.com",
+                "Google",
+                null,
+            ),
+        ),
+        modifier = Modifier.background(Color.White),
+    ) { }
```

Comment:
Generally, we try to keep the click lambdas as named parameters in previews to help keep the code concise and make future parameter refactors easier.
```suggestion
        modifier = Modifier.background(Color.White),
        onAccountClick = {},
    )
```